resources/smpp/ra/src/main/java/org/restcomm/slee/resource/smpp/SmppSessionsImpl.java

```diff
@@ -495,4 +492,113 @@ public void fireUnknownThrowable(Throwable throwable) {
 
 	}
 
+	public class SenderThread extends Thread {
+
+		private Boolean running = true;
+		private LinkedBlockingQueue<SmppSendingTask> queue = new LinkedBlockingQueue<>();
+
+		public SenderThread(String name) {
+			super(name);
+		}
+
+		public void deactivate() {
+			running = false;
```

Comment:
I guess its a matter of flavor. Thread will get stopped only when session is terminated. So its really rear case.